src/core/server/saved_objects/routes/delete.ts

```diff
@@ -31,11 +31,20 @@ export const registerDeleteRoute = (router: IRouter, { coreUsageData }: RouteDep
     router.handleLegacyErrors(async (context, req, res) => {
       const { type, id } = req.params;
       const { force } = req.query;
+      const { getClient, typeRegistry } = context.core.savedObjects;
 
       const usageStatsClient = coreUsageData.getClient();
       usageStatsClient.incrementSavedObjectsDelete({ request: req }).catch(() => {});
 
-      const result = await context.core.savedObjects.client.delete(type, id, { force });
+      const includedHiddenTypes = [type].filter(
+        (supportedType) =>
+          typeRegistry.isHidden(supportedType) &&
+          typeRegistry.isImportableAndExportable(supportedType)
```

Comment:
Should we allow deleting non importable/exportable hidden types?

Reply:
I don't think we should be impacting the `/delete` route, or do you see a reason to do so?